File: jambeth/jAmbeth.Merge/src/de/osthus/ambeth/proxy/AbstractEntityFactory.java
package de.osthus.ambeth.proxy;

import de.osthus.ambeth.ioc.annotation.Autowired;
import de.osthus.ambeth.log.ILogger;
import de.osthus.ambeth.log.LogInstance;
import de.osthus.ambeth.merge.IEntityFactory;
import de.osthus.ambeth.merge.IEntityMetaDataProvider;
import de.osthus.ambeth.merge.model.IEntityMetaData;

public abstract class AbstractEntityFactory implements IEntityFactory
{
	@SuppressWarnings("unused")
	@LogInstance
	private ILogger log;

	@Autowired
	protected IEntityMetaDataProvider entityMetaDataProvider;

	@SuppressWarnings("unchecked")
	@Override
	public <T> T createEntity(Class<T> entityType)
	{
		IEntityMetaData metaData = entityMetaDataProvider.getMetaData(entityType);
		return (T) createEntity(metaData);
	}

	@Override
	public boolean supportsEnhancement(Class<?> enhancementType)
	{
		return false;
	}
}

File: ambeth/Ambeth.Merge/ambeth/proxy/AbstractEntityFactory.cs
using System;
using De.Osthus.Ambeth.Ioc.Annotation;
using De.Osthus.Ambeth.Log;
using De.Osthus.Ambeth.Merge;
using De.Osthus.Ambeth.Merge.Model;

namespace De.Osthus.Ambeth.Proxy
{
    public abstract class AbstractEntityFactory : IEntityFactory
    {
	    [LogInstance]
        public ILogger Log { private get; set; }

	    [Autowired]
	    public IEntityMetaDataProvider EntityMetaDataProvider { protected get; set; }

	    public T CreateEntity<T>()
	    {
            return (T) CreateEntity(typeof(T));
	    }

        public virtual Object CreateEntity(Type entityType)
        {
            IEntityMetaData metaData = EntityMetaDataProvider.GetMetaData(entityType);
            return CreateEntity(metaData);
        }

        public abstract Object CreateEntity(IEntityMetaData metaData);

	    public virtual bool SupportsEnhancement(Type enhancementType)
	    {
		    return false;
	    }
    }
}
